promql/parse.go

```diff
@@ -323,9 +326,15 @@ func (p *parser) recover(errp *error) {
 	e := recover()
 	if e != nil {
 		if _, ok := e.(runtime.Error); ok {
-			panic(e)
+			// Print the stack trace but do not inhibit the running application.
+			buf := make([]byte, 64<<10)
+			buf = buf[:runtime.Stack(buf, false)]
```

Comment:
So `fmt.Printf` seems to do hide trailing null bytes. Regarding memory this still means `buf` is `64 << 10` bits large even though the final content is way less.
Technically this is the correct solution that correctly frees unused memory, even if it's not relevant for this purpose.